### ISSUE TITLE
Link feed field recommended, but not required for Atom and JSON feeds

### ISSUE THREAD

**Issue by pruser** [gorilla/feeds#52]
I noticed "runtime error: invalid memory address or nil pointer dereference" while using feeds library, so i started checking the issue.
It turned out that the library assumes that Feed.Link field is set although it has pointer type.

After checking the specs:
The Link field is required only for RSS feeds.
Atom feeds does not require feed to contain <link> element, but it's recommended though.
JSON feed spec state that home_page_url and feed_url is optional, but strongly recommended.

I can fix this issue for Atom and JSON feeds, but I'd like to know your opinion first.

Would you like to keep it as it is, or to create non required fields only if Feed.Link is set?

specs:
https://validator.w3.org/feed/docs/atom.html
https://jsonfeed.org/version/1

**Comment by kisielk** [comment]
If you could provide a PR or reproducible example, that would be helpful. Thanks!

**Comment by pruser** [comment]
I just realised that gorilla feeds use Feed.Link field for generating Atom channel Id according to the definition of Id field.

**Comment by preslavrachev** [comment]
What is the state of this issue? I just recently came to the same problem.

**Comment by pruser** [comment]
Even though the 'link' element is not required by Atom feed spec the 'id' is. The purpose of 'id' field is to contain unique and permanent URI, so I think it can be used to create target feed object from gorilla 'Feed' object.

I modified conversion logic to return error instead of crashing on null dereference. 
https://github.com/gorilla/feeds/pull/61 . 
The PR was not accepted by maintainers and was closed automatically.
